r/tools/nixlibs.R

```diff
@@ -168,17 +178,22 @@ select_binary <- function(os = tolower(Sys.info()[["sysname"]]),
   } else {
     # No binary available for arch
     cat(sprintf("*** Building on %s %s\n", os, arch))
-    NULL
+    binary <- NULL
   }
+  return(binary)
 }
 
 # This tests that curl and OpenSSL are present (bc we can include their headers)
 # and it checks for other versions/features and raises errors that we grep for
-test_for_curl_and_openssl <- "
-#include <ciso646>
+test_for_curl_and_openssl <-
+  "#include <ciso646>
+#ifndef __APPLE__
 #ifdef _LIBCPP_VERSION
 #error Using libc++
 #endif
+#elif __GLIBCXX__
+#error Using libstdc++
```

Comment:
Did you confirm that this is necessary? 

If so, I might refactor this check for readability: ifdef apple, then check for libstdcxx, else check for libcxx.

Reply:
It is theoretically possible but very unlikely with no support from apple and brew also defaulting to libc++, so I removed it.